### PR TITLE
Add build matrix for D8 and D9 testing

### PR DESCRIPTION
**Pull request by mglaman**
Adds a build matrix for ^8.9 && Webform 5.x and ^9.0 && Webform 6.x

### CODE REVIEW

**Comment by mglaman** [comment]
The tests are failing on D9 due to the `civicrm.install` file:

```
  $installed = $setup->checkInstalled();
  if ($installed->isSettingInstalled() || $installed->isDatabaseInstalled()) {
    drupal_set_message(t("CiviCRM appears to have already been installed. Skipping full installation."));
  }
```

**Comment by mglaman** [comment]
That was fixed by https://github.com/civicrm/civicrm-drupal-8/pull/53. So once `civicrm-drupal-8` gets a release we'll be unblocked.

**Comment by mglaman** [comment]
We should be able to fix by testing with the latest RC. So we'll just need to use `5.33.x-dev` as the version constraint. See https://packagist.org/packages/civicrm/civicrm-core#5.33.x-dev

**Comment by mglaman** [comment]
😭 The silent failure of `drupal_set_message` is a violent failure now.

```
Exception: CiviCRM appears to have already been installed. Skipping full installation.
```

**Comment by mglaman** [comment]
Locally I broke it apart, and it seems to fail on the database check. The `isSettingsInstalled` check doesn't error.

```
  $installed = $setup->checkInstalled();
  if ($installed->isSettingInstalled()) {
    throw new \Exception("CiviCRM settings already exist. Skipping full installation.");
  }
  if ($installed->isDatabaseInstalled()) {
    throw new \Exception("CiviCRM database is installed. Skipping full installation.");
  }
```

Result:

```
1) Drupal\Tests\webform_civicrm\FunctionalJavascript\ContactSubmissionTest::testSubmitWebform with data set #0 ('Individual', array(array('Frederick', 'Pabst')))
Exception: CiviCRM database is installed. Skipping full installation.
```

**Comment by mglaman** [comment]
So this makes sense. Drupal uses the same database for each test but different prefixes. CiviCRM does not support prefixes, as `'SHOW TABLES LIKE "civicrm_%"'` shows. But that should only be a problem locally if the site was installed. In the CI we don't have an installed site.

**Comment by mglaman** [comment]
🤦 oh! of course. The first test passes but then subsequent ones fail due to the lack of cleanup of the tables:

```
.EEEEESS.                                                           9 / 9 (100%)
```

File: tests/src/FunctionalJavascript/CiviCrmTestBase.php [hunk 1]
```diff
@@ -28,14 +28,36 @@ protected function setUp() {
...
+    $civicrm_test_conn->query('SET FOREIGN_KEY_CHECKS = 0;')->execute();
+    $civicrm_schema = $civicrm_test_conn->schema();
+    $tables = $civicrm_schema->findTables('%');
+    foreach ($tables as $table) {
+      if ($civicrm_schema->dropTable($table)) {
+        unset($tables[$table]);
+      }
+    }
+    $civicrm_test_conn->query('SET FOREIGN_KEY_CHECKS = 1;')->execute();
+  }
```
Comment (mglaman) [inline tests/src/FunctionalJavascript/CiviCrmTestBase.php:47]: I'm going to cherry-pick this into a new PR. Turns out we were not dropping any of the `civicrm_` tables between tests.

**Comment by mglaman** [comment]
D9 failure:

```
User deprecated function: Deprecated function ::civicrm_api3_contribution_transact, use The contibution.transact api is unsupported & known to have issues. Please see the section at the bottom of https://docs.civicrm.org/dev/en/latest/financial/OrderAPI/ for getting off it. Array ( [civi.tag] => deprecated ) in CRM_Core_Error_Log->log() (line 58 of /home/runner/drupal/vendor/civicrm/civicrm-core/CRM/Core/Error/Log.php).
CRM_Core_Error_Log->log('warning', 'Deprecated function ::civicrm_api3_contribution_transact, use The contibution.transact api is unsupported &amp; known to have issues. Please see the section at the bottom of https://docs.civicrm.org/dev/en/latest/financial/OrderAPI/ for getting off it.
Array
(
    [civi.tag] =&gt; deprecated
)
', Array) (Line: 85)
Psr\Log\AbstractLogger->warning('Deprecated function ::civicrm_api3_contribution_transact, use The contibution.transact api is unsupported &amp; known to have issues. Please see the section at the bottom of https://docs.civicrm.org/dev/en/latest/financial/OrderAPI/ for getting off it.', Array) (Line: 1042)
CRM_Core_Error::deprecatedFunctionWarning('The contibution.transact api is unsupported &amp; known to have issues. Please see the section at the bottom of https://docs.civicrm.org/dev/en/latest/financial/OrderAPI/ for getting off it') (Line: 42)
civicrm_api3_contribution_transact(Array) (Line: 89)
Civi\API\Provider\MagicFunctionProvider->invoke(Array) (Line: 150)
Civi\API\Kernel->runRequest(Array) (Line: 81)
Civi\API\Kernel->runSafe('contribution', 'transact', Array) (Line: 22)
civicrm_api('contribution', 'transact', Array) (Line: 539)
wf_civicrm_api('contribution', 'transact', Array) (Line: 1825)
wf_crm_webform_postprocess->submitLivePayment() (Line: 141)
wf_crm_webform_postprocess->validate(Array, Object, Object) (Line: 136)
Drupal\webform_civicrm\Plugin\WebformHandler\CivicrmWebformHandler->validateForm(Array, Object, Object) (Line: 2618)
Drupal\webform\Entity\Webform->invokeHandlers('validateForm', Array, Object, Object) (Line: 1733)
Drupal\webform\WebformSubmissionForm->validateForm(Array, Object)
call_user_func_array(Array, Array) (Line: 82)
Drupal\Core\Form\FormValidator->executeValidateHandlers(Array, Object) (Line: 273)
Drupal\Core\Form\FormValidator->doValidateForm(Array, Object, 'webform_submission_civicrm_webform_test_add_form') (Line: 118)
Drupal\Core\Form\FormValidator->validateForm('webform_submission_civicrm_webform_test_add_form', Array, Object) (Line: 589)
Drupal\Core\Form\FormBuilder->processForm('webform_submission_civicrm_webform_test_add_form', Array, Object) (Line: 321)
Drupal\Core\Form\FormBuilder->buildForm(Object, Object) (Line: 48)
Drupal\Core\Entity\EntityFormBuilder->getForm(Object, 'add') (Line: 1195)
Drupal\webform\Entity\Webform->getSubmissionForm() (Line: 76)
Drupal\webform\Controller\WebformEntityController->addForm(Object, Object)
call_user_func_array(Array, Array) (Line: 123)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 573)
Drupal\Core\Render\Renderer->executeInRenderContext(Object, Object) (Line: 124)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->wrapControllerExecutionInRenderContext(Array, Array) (Line: 97)
Drupal\Core\EventSubscriber\EarlyRenderingControllerWrapperSubscriber->Drupal\Core\EventSubscriber\{closure}() (Line: 158)
Symfony\Component\HttpKernel\HttpKernel->handleRaw(Object, 1) (Line: 80)
Symfony\Component\HttpKernel\HttpKernel->handle(Object, 1, 1) (Line: 57)
Drupal\Core\StackMiddleware\Session->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\KernelPreHandle->handle(Object, 1, 1) (Line: 106)
Drupal\page_cache\StackMiddleware\PageCache->pass(Object, 1, 1) (Line: 85)
Drupal\page_cache\StackMiddleware\PageCache->handle(Object, 1, 1) (Line: 47)
Drupal\Core\StackMiddleware\ReverseProxyMiddleware->handle(Object, 1, 1) (Line: 52)
Drupal\Core\StackMiddleware\NegotiationMiddleware->handle(Object, 1, 1) (Line: 23)
Stack\StackedHttpKernel->handle(Object, 1, 1) (Line: 706)
Drupal\Core\DrupalKernel->handle(Object) (Line: 19)
```

**Comment by mglaman** [comment]
D9 is failing due to a CiviCRM deprecation. The docs link is broken, so I opened an issue: https://lab.civicrm.org/dev/core/-/issues/2235. I also opened an issue for this project https://www.drupal.org/project/webform_civicrm/issues/3186875

**Comment by mglaman** [comment]
Since we need to sort out the Transact API deprecation (either fix or why is it _so loud_) I spun out two fixes here into their own PR:

* https://github.com/colemanw/webform_civicrm/pull/402
* https://github.com/colemanw/webform_civicrm/pull/403

**Comment by mglaman** [comment]
I wonder if it's something in Drupal 9...

**Comment by mglaman** [comment]
@KarinG so this is green because I commented out the error message check. Locally I cannot get the error message to pop.

I guess we could close #402 and #403 to just merge this. Or use those and make a more cleaner PR from this one.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -8,7 +8,16 @@ on:
...
-    name: PHPUnit tests
+    strategy:
+      matrix:
+        include:
+          - drupal: '^8.9'
+            civicrm: '~5.32'
+          - drupal: '^8.9'
+            civicrm: '5.33.x-dev'
+          - drupal: '^9.0'
+            civicrm: '5.33.x-dev'
```
Comment (mglaman) [inline .github/workflows/main.yml:19]: Now we can test various versions of CiviCRM and Drupal minor releases much easier. We could port this to a new PR to just get `^8.9`  testing with 5.32 and 5.33RC

Comment (KarinG) [reply .github/workflows/main.yml:19]: That's really 👍 helpful!

**Comment by KarinG** [comment]
Let's just merge this! 

On my list: to ask on our end and get input as to why a deprecation notice is bubbling up as a message/error rather than just quietly going into Drupal's logging channels.

**Pull request merged**